driver/bpf/probe.c

```diff
@@ -86,34 +128,79 @@ BPF_PROBE("raw_syscalls/", sys_enter, sys_enter_args)
 	if (stash_args(stack_ctx.args))
 		return 0;
 
-	call_filler(ctx, &stack_ctx, evt_type, drop_flags);
+	call_filler(ctx, &stack_ctx, evt_type, drop_flags, socketcall_syscall_id);
 #endif
 	return 0;
 }
 
 BPF_PROBE("raw_syscalls/", sys_exit, sys_exit_args)
 {
-	const struct syscall_evt_pair *sc_evt;
-	ppm_event_code evt_type;
-	int drop_flags;
-	long id;
-	bool enabled;
-	struct scap_bpf_settings *settings;
-	long retval;
-
-	if (bpf_in_ia32_syscall())
-		return 0;
+	const struct syscall_evt_pair *sc_evt = NULL;
+	ppm_event_code evt_type = -1;
+	int drop_flags = 0;
+	long id = 0;
+	bool enabled = false;
+	struct scap_bpf_settings *settings = 0; 
+	long retval = 0;
+	int socketcall_syscall_id = -1;
+
+#ifdef __NR_socketcall
+	socketcall_syscall_id = __NR_socketcall;
+#endif
 
 	id = bpf_syscall_get_nr(ctx);
 	if (id < 0 || id >= SYSCALL_TABLE_SIZE)
 		return 0;
 
-#if defined(CAPTURE_SOCKETCALL) && defined(BPF_SUPPORTS_RAW_TRACEPOINTS)
-	if(id == __NR_socketcall)
+	if (bpf_in_ia32_syscall())
 	{
-		id = convert_network_syscalls(ctx);
-	}
+		if (id == __NR_ia32_socketcall)
+		{
+			socketcall_syscall_id = __NR_ia32_socketcall;
+		}
+		else
+		{
+#if defined(CONFIG_X86_64) && defined(CONFIG_IA32_EMULATION)
+			/*
+			 * When a process does execve from 64bit to 32bit, TS_COMPAT is marked true
+			 * but the id of the syscall is __NR_execve, so to correctly parse it we need to
+			 * use 64bit syscall table. On 32bit __NR_execve is equal to __NR_ia32_oldolduname
+			 * which is a very old syscall, not used anymore by most applications
+			 */
+#ifdef __NR_execveat
+			if(id != __NR_execve && id != __NR_execveat)
```

Comment:
maybe we could add a test for it on 32 bit

Reply:
Added in 7ed2ef7da6a518af02d7298791e38334db0d54c0